### ISSUE TITLE
`format_push_string` fix suggestion causes `clippy::write_with_newline` when the `format!` literal ends with a newline

### ISSUE THREAD

**Issue by doonv** [rust-lang/rust-clippy#16841]
### Summary

The fix suggestion for `clippy::format_push_string` should use `writeln!` when the `format!` literal ends with a newline. Currently the suggestion causes `clippy::write_with_newline`

### Reproducer

Code:
```rust
let mut s = String::new();
s += &format!("0x{:X}\n", 1024);
```

Current output:
```rust
let mut s = String::new();
let _ = write!(s, "0x{:X}\n", 1024);
```
This also causes another Clippy lint:
```
warning: using `write!()` with a format string that ends in a single newline
 --> src/main.rs:6:13
  |
6 |     let _ = write!(s, "0x{:X}\n", 1024);
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_with_newline
  = note: `#[warn(clippy::write_with_newline)]` on by default
help: use `writeln!` instead
  |
6 -     let _ = write!(s, "0x{:X}\n", 1024);
6 +     let _ = writeln!(s, "0x{:X}", 1024);
  |
```

Desired output:
```rust
let mut s = String::new();
let _ = writeln!(s, "0x{:X}", 1024);
```

### Version

```text
rustc 1.96.0-nightly (48cc71ee8 2026-03-31)
binary: rustc
commit-hash: 48cc71ee88cd0f11217eced958b9930970da998b
commit-date: 2026-03-31
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.2
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by ada4a** [comment]
Clippy lints are supposed to be composable, so it's perfectly fine for a fix from one lint to unlock a further fix from another lint. In fact, `cargo clippy --fix` takes advantage of just this, by repeatedly applying fixes until there are no more. Therefore, I'd close this as wontfix -- but don't hesitate to re-open if there's anything I'm missing.
